src/components/Board.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import Square from './Square';
+import BoardContext from '../context/BoardContext'
+
+function Board(props) {
+    const { board, setBoard, turn, setTurn, history, setHistory } = useContext(BoardContext);
+
+    function declareWinner(board) {
+        const winningCombos = [
+            [3, 4, 5],
+            [0, 1, 2],
+            [0, 3, 6],
+            [2, 5, 8],
+            [1, 4, 7],
+            [0, 4, 8],
+            [2, 4, 6],
+            [6, 7, 8],
+        ];
+
+        function threeInARow() {
+            for (let i = 0; i < winningCombos.length; i++) {
+                const [first, second, third] = winningCombos[i];
+                // When you get 3 in a row, in any direction.
+                if (board[first] && board[first] === board[second] && board[first] === board[third]) {
+                    return board[first];
+                }
+            }
+            return null;
+        }
+        threeInARow()
+    };
+
+    const winner = declareWinner(board);
+
+    // Check for winner, every move.
+    let status = null;
+    if (winner) {
+        status = `${winner} WON THE GAME!`;
+    };
+
+    function timeTravel(turn) {
+            const boardInHistory = [...history];
+            setBoard(boardInHistory[turn]);
+    }
+    
+    function clickHandler(i) {
+        if (!winner && !board[i]) {
+            const newBoard = [...board];
+            const newHistory = [...history];
+            newBoard[i] = turn;
+            setBoard(newBoard);
+            setHistory(newHistory);
+            setTurn((turn === 'X') ? 'O' : 'X');
+        };
+    }
+
+    const moves = history.forEach(square, index => { // use map instead ?
+        const steps = index ? `Go to move ${index}` : 'Restart Game';
+        return (
+            <li key={index}>
+                <button onClick={() => timeTravel(index)}>{steps}</button>
+            </li>
+        )
+    });
+
+    return (
+        <div className='game'>
+            <div className="board">
+                <h4>Currently player {turn}'s turn. Make your move.</h4>
+                <h3>{status}</h3>
+                <h3>{moves}</h3>
+
+                <div className="row board-row">
+                    <Square onClick={() => clickHandler(0)} value={board[0]} />
```

Comment:
You can shorten this up a bit by using the `chunk` method from lodash if you like, or by implementing your own chunk method! https://lodash.com/docs/4.17.15#chunk

```js
{_.chunk(board, 3).map(row => (
  <div className="row board-row">{row.map((cell, i) => <Square onClick={() => clickHandler(i)} value={cell} }</div>
))}
```